### ISSUE TITLE
RazerS3 error concerning length of fastq quality string

### ISSUE THREAD

**Issue by bio15anu** [seqan/seqan#2315]
Hi all,

I am testing RazerS3 with my data as it seems to be returning far fewer alignments than expected. I found that with many reads I am getting this error: 

`/home/osboxes/seqan/include/seqan/basic/basic_exception.h:363 FAILED!  (Uncaught exception of type seqan::ParseError: Fastq quality string is expected to be of the same length as the sequence! But was not.)`

However, this was returned for example when running the alignment on only the following PE reads:

`@3_AZNP01000387.1:258181-258645`
`TCCACAGATGTCAAATAGATGTGTCCTCTAGTTCCCTCCAGTCGTCGAGATCATAAATTCTAGGGAAATTGTGGGTAGGGACAGTTCATTATTTTCTTGCAACAGTTTGTG`
`+`
`HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGGGGGGGGGGGGGGGFFFFFFFFEEEEEEEDDDDDCCCCCBBBAAAA@@@???>>>===<<;;:::998877655`

`@3_AZNP01000387.1:258181-258645`
`CTTAAAGAAACAAATGGTTCTCTCTGTCTCCTTAGGTGAGAGTAAGGACCGTTCTTTAAATACAAAATATATAAATACAAATTAGGAGAACCAAATAATGCACAACTTTTG`
`+`
`HHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGGGGGGGGGGGGGGGFFFFFFFFEEEEEEEDDDDDCCCCCBBBAAAA@@@???>>>===<<;;:::998877655`

The command I am using to run the alignment is below:
`razers3 -rr 100 -i 95 -tc 1 -dr 0 -m 5 -pa -ds -o test.bam genome.fa read_1.fastq read_2.fastq `

Best regards,
Adam

**Comment by rrahn** [comment]
Thanks for reporting this. We'll check this.
Which version of Razers3 did you use?

**Comment by rrahn** [comment]
I could reproduce the error with razers3 and happens while reading the first record of the first fastq file in the `estimateReadLength` function.
Strangely, in a regression unit test the error does not occur and the fastq file is parsed correctly.
We'll investigate this.

**Comment by bio15anu** [comment]
Hmm that's a puzzle! Just so I'm clear does that mean it worked correctly in a previous version? Would you be able to advise me of the latest software version where the problem does not occur? To give some additional information, we originally ran into the problem with software version 3.5.3 although it failed without reporting any error. I think actually it worked with parameter `-i 90` but not with `-i 95`, which was strange because the read is expected to align with >99% identity.

**Comment by bio15anu** [comment]
Sorry no I just retested on version 3.5.3 and we get the same error, regardless of the identity parameter.

**Comment by rrahn** [comment]
No, the exception is thrown, while extracting the length of the first record in the read1.fastq file. 
As far as I could see, this is completely independent of the program options to Razers3. I just need to figure out, whether the library function is buggy, or wether the provided input from Razers3 is kind of corrupted.

**Comment by bio15anu** [comment]
Hi rrahn, I think this error with one read is a separate issue to what I was having before when I was getting less than the expected number of alignments. I adjusted the `-ll` and the `-le` parameter to better fit my data and now the read is mapping when I include it with all the others. The error is still thrown when I try to map the read individually, which I suspect is due to the @ symbols in the quality string. Could this exception simply be thrown due to RazerS3 deciding to parse the read on it's own as if it was in multi-line fastq format?
